### DISCUSSION TITLE
Mid size file decompression: match offset (131072) bigger than current history (0)

### DISCUSSION THREAD

**Discussion by khbence** [klauspost/compress#738] [Q&A]
Hi, I'm trying to use the decode functionality, I'm working in a container environment so strange to say that it worked at a different computer but not on another. (same architecture).

`	f, err := os.Open("localTest.txt")
	if err != nil {
		panic(err)
	}
	d, err := zstd.NewReader(nil, zstd.WithDecoderConcurrency(4))
	if err != nil {
		panic(err)
	}
	if err := d.Reset(f); err != nil {
		panic(err)
	}
	stream := d.IOReadCloser()
	data, err := io.ReadAll(stream)
	if err != nil {
		panic(err)
	}
	cytolog.Info("%v", data)`
But when I try to decompress the attached file it gives the error message in the title. A generate these files by compressing (with similar GO zstd code) a quite repetitive JSON string. If I do the same with a smaller file it works fine. Sometimes it can also produce "litLengths:sequence decoder not defined" but hard to reproduce, in debug mode it also behaves somewhat differently, so I guess it's some datarace.
[localTest.txt](https://github.com/klauspost/compress/files/10384415/localTest.txt)

**Comment by klauspost** [comment]
The file you provided is corrupt (added an extension):
```
λ zstd -d localTest.txt.zst
localTest.txt.zst    : 0 MB...     localTest.txt.zst : Decoding error (36) : Corrupted block detected
```
Be sure you didn't transfer the file as text and had CR changed to CR+LF.
